File: utils_py/list.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def is_equal_list(l1: ListNode, l2: ListNode):
    if not l1 and not l2:
        return True
    if not l1 or not l2:
        return False
    if l1.val != l2.val:
        return False
    return is_equal_list(l1.next, l2.next)

def print_list(lst: ListNode):
    p = lst
    while p:
        print(p.val, end=', ')
        p = p.next

File: utils_js/list.js

function ListNode(val, next) {
  this.val = (val===undefined ? 0 : val);
  this.next = (next===undefined ? null : next);
}

function isEqualList(l1, l2) {
  let p1 = l1;
  let p2 = l2;
  while (p1 && p2) {
    if (p1.val != p2.val)
      return false;
    p1 = p1.next;
    p2 = p2.next;
  }
  if (p1 || p2)
    return false;
  return true;
}

function printList(list) {
  let p = list;
  while (p) {
    console.log(p.val);
    p = p.next;
  }
}

const ListUtils = {
  isEqualList: isEqualList,
  printList: printList,
};

export { ListUtils, ListNode };